react/hooks/useAppSettings.ts

```diff
@@ -0,0 +1,33 @@
+import { useQuery } from 'react-apollo'
+import { useEffect, useState } from 'react'
+
+import GET_SETTINGS from '../queries/getSettings.graphql'
+
+const DEFAULT_DECIMALS = 2
+const DEFAULT_PRICES_WITH_TAX = false
+
+interface Settings {
+  decimals: number
+  pricesWithTax: boolean
+}
+
+const useAppSettings = (): Settings => {
+  const [settings, setSettings] = useState<Settings>({
+    decimals: DEFAULT_DECIMALS,
+    pricesWithTax: DEFAULT_PRICES_WITH_TAX,
+  })
+
+  const { data, loading } = useQuery(GET_SETTINGS, { ssr: false })
+
+  useEffect(() => {
+    if (!loading) {
+      const { decimals, pricesWithTax } = JSON.parse(data?.appSettings?.message)
+
+      if (decimals && pricesWithTax) setSettings({ decimals, pricesWithTax })
```

Comment:
This will fail if `pricesWithTax` is `false`, which is not the desired behavior. Also, if `decimals` is `0`, it will also fail.